assets/config.php

```diff
@@ -178,6 +177,37 @@
         'drop_tenant_databases_on_migrate_fresh' => false,
     ],
 
+    /**
+     * Requires Postgres with single-database tenancy.
+     */
+    'rls' => [
+        /**
+         * Scope tenant models using RLS.
+         */
+        'enabled' => false,
+
+        /**
+         * Permissions to grant to the tenant Postgres users.
+         *
+         * By default, all permissions are granted.
+         *
+         * @see Stancl\Tenancy\Jobs\CreatePostgresUserForTenant
+         */
+        'user_permissions' => ['ALL'],
```

Comment:
Just to double check: Do these permissions only apply on the tenant's schema, or will the created user have these permissions across the whole postgres server?

If it's the latter, then we might want to use something more restrictive by default.

Reply:
Looks like they apply on the tenant's schema only https://github.com/archtechx/tenancy/commit/d005e7b06dbfdd4b81192f0dfd51faabb7ce7097

Reply:
Also, updated the default permissions to `'UPDATE', 'DELETE', 'SELECT', 'INSERT'`, which is a better default since it only allows the basic operations

Reply:
> What's the reason for this one not being needed?

Because I couldn't figure this one out myself, I asked ChatGPT (the prompt was "A Postgres DB user I created doesn't have the 'CONNECT' privilege, though it can still connect to the Postgres database. Why exactly does this happen?")

The reply:

> In PostgreSQL, the ability to connect to a database is controlled by two main factors: the server configuration and the database user privileges.
> 
> By default, PostgreSQL allows all users to connect to a database, even if they don't have explicit privileges granted. This behavior is determined by the server configuration parameter `pg_hba.conf` (Host-Based Authentication Configuration). This file defines the authentication rules that the PostgreSQL server uses to determine whether a client can connect.
> 
> When a client attempts to connect, the server consults the `pg_hba.conf` file to check if the client's IP address and authentication method match any of the rules. If a match is found, the client is allowed to connect, regardless of the specific user privileges within the database.
> 
> Once the client has successfully connected to the database, the user's privileges within the database itself determine what actions they can perform. The privileges control access to tables, views, functions, and other database objects.
> 
> Therefore, even if a database user doesn't have the 'CONNECT' privilege explicitly granted, they can still connect to the database as long as the server's authentication rules in `pg_hba.conf` allow it. However, the user's actions within the database will be restricted based on their assigned privileges.

Reply:
So in conclusion, by default, the CONNECT privilege is granted by Postgres by default depending on the user's Postgres config (and other commonly granted privileges; for convenience).

We can either strip the newly created Postgres user of all its privileges and give it only the privileges configured in our package's config, or let things as they are now. The first option is definitely the more secure one, since the user would have complete control over the permissions of the tenant Postgres users.

Reply:
> the CONNECT privilege is granted to new Postgres users by default depending on the user's Postgres config

Revisting this now, and this doesn't seem to be true. The ability to connect to the db itself depends on the config – no privilege is granted

Reply:
With the default Postgres configuration, the Postgres users don't need the `CONNECT` privilege to connect to the db. The privilege is only needed if you're trying to connect to the db from a non-trusted IP, and by default, the `pg_hba.conf` allows all loopback connections.

So the current default privileges are probably good enough – if someone uses a different config, he should revise the privileges, and customize them if needed.